integrated_channels/canvas/exporters/content_metadata.py

```diff
@@ -0,0 +1,56 @@
+"""
+Content metadata exporter for Canvas
+"""
+
+from logging import getLogger
+
+from integrated_channels.integrated_channel.exporters.content_metadata import ContentMetadataExporter
+
+LOGGER = getLogger(__name__)
+
+
+class CanvasContentMetadataExporter(ContentMetadataExporter):
+    """
+        Canvas implementation of ContentMetadataExporter.
+    """
+    DATA_TRANSFORM_MAPPING = {
+        'name': 'title',
+        'course_code': 'number',
+        'start_at': 'start',
+        'end_at': 'end',
+        'integration_id': 'key',
+        'syllabus_body': 'description',
+        'default_view': 'default_view',
+    }
+
+    LONG_STRING_LIMIT = 2000
+
+    def transform_description(self, content_metadata_item):
+        """
+        Return the course description and enrollment url as Canvas' syllabus body attribute.
+        This will display in the Syllabus tab in Canvas.
+        """
+        enrollment_url = content_metadata_item.get('enrollment_url', None)
+        base_description = "<a href={enrollment_url}>To edX Course Page</a><br />".format(
+            enrollment_url=enrollment_url)
+        full_description = content_metadata_item.get('full_description') or None
+        if full_description and len(full_description + enrollment_url) <= self.LONG_STRING_LIMIT:
+            description = "{base_description}{full_description}".format(
+                base_description=base_description,
+                full_description=full_description
+            )
+        else:
+            short_description = content_metadata_item.get(
+                'short_description', content_metadata_item.get('title', '')
+            )
+            description = "{base_description}{short_description}".format(
+                base_description=base_description, short_description=short_description
+            )
+
+        return description
+
+    def transform_default_view(self, content_metadata_item):  # pylint: disable=unused-argument
```

Comment:
not sure how we will use this field, but does it need the `content_metadata_item` argument? it looks like it is not used right now

Reply:
I think this is due to how the base class interprets these methods. The other exporters that make up fields only used in the target system (I think Success Factors was one I saw or Cornerstone...cannot remember) also have this unused argument.

> not sure how we will use this field

it's required by Canvas. This is the field that tells Canvas we want the "Home" page to be the syllabus. It needs to be sent along with the rest of the fields in the mapping.